daemon/api.go

```diff
@@ -1189,6 +1190,14 @@ func postSnap(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 	inst.ctx = r.Context()
 
+	if inst.Channel != "" {
```

Comment:
this come from snapRevisionOptions, so we need a check also in api_download.go , maybe we want a helper on snapRevisionOptions ? like validate() ?